src/libsyntax/tokenstream.rs

```diff
@@ -491,8 +494,7 @@ pub struct Cursor(CursorKind);
 #[derive(Clone)]
 enum CursorKind {
     Empty,
-    Tree(TokenTree, bool /* consumed? */),
-    JointTree(TokenTree, bool /* consumed? */),
+    Tree(TokenTree, IsJoint, bool /* consumed? */),
```

Comment:
Similar note on field placement as https://github.com/rust-lang/rust/pull/56964/#discussion_r242751346 applies here. Maybe this is relying on field reordering packing everything in that padding between the discriminant and the fields of each variant?